### PR TITLE
Clicking on a saved TextDetail opens it again. 

### PR DESCRIPTION
**Pull request by hdeweirdt**
Includes quite a rewrite of the DetailFragment, which is now a DetailFragmentWithEmotionAvatar (longer name but a clearer distinction with the Text/Photo/...Fragment. 
The rewrite was to ensure that opening an existing detail is made transparent. Just pass a detail and the correct Fragment will open. 

TextFragment can now be passed the UUID of an existing TextDetail and it will read the contents from it (new UC). 
When saving a TextDetail in the EventVM you can now pass an optional existingDetail parameter. The UC handles the correct action (overwrite existing file or create new Detail).

### CODE REVIEW

File: .idea/misc.xml [hunk 1]
```diff
@@ -39,7 +39,7 @@
       </value>
     </option>
   </component>
-  <component name="ProjectRootManager" version="2" languageLevel="JDK_1_8" project-jdk-name="1.8" project-jdk-type="JavaSDK">
+  <component name="ProjectRootManager" version="2" languageLevel="JDK_1_7" project-jdk-name="1.8" project-jdk-type="JavaSDK">
```
Comment (eothein) [inline .idea/misc.xml]: Why is this changed?

**Comment by eothein** [comment]
Hello Harm

Quite some work done! From my perspective this seems to be all right, but I have to say that it is a lot of code to review and to keep  the overview. 

I see that the Fragment creation is now delegated to the Activity which I think is indeed better. Depending on the detail type it creates the correct Fragment, which makes it more scalable i.e. in the case where ortho wants a new type of detail. 

If you have any specific questions, could you direct me to the commit, class and line numbers. This way I could provide better and targetted feedback.

File: .idea/misc.xml (hunk 1 above)
Comment (hdeweirdt) [reply .idea/misc.xml]: My bad, that's a local setting that sometimes slip in when working on my home PC. Your AS will automatically change that back to 1_8 normally.
